src/encoded/types/file.py

```diff
@@ -136,12 +150,19 @@ def __name__(self):
             return self.uuid
         return properties.get(self.name_key, None) or self.uuid
 
+
     def unique_keys(self, properties):
         keys = super(File, self).unique_keys(properties)
         if properties.get('status') != 'replaced':
             if 'md5sum' in properties:
                 value = 'md5:{md5sum}'.format(**properties)
-                keys.setdefault('alias', []).append(value)
+                if self.registry[CONNECTION].get_by_unique_key('alias', value):
```

Comment:
Why don't you try to set the value and then catch the exception if it already exists. Then determine whether to raise the exception or continue based on if the file is md5sum constrained.

Reply:
I don't think the exception can be caught from here, as the keys go into postgres later